pygmt/tests/test_clib_loading.py

```diff
@@ -64,18 +89,85 @@ def test_load_libgmt_with_a_bad_library_path(monkeypatch):
+            return libname
+        if isinstance(libname, str):
+            # libname is an invalid library path in str type,
+            # raise OSError like the original ctypes.CDLL
+            raise OSError(f"Unable to find '{libname}'")
+        # libname is a loaded GMT library
+        return loaded_libgmt
+
+    with monkeypatch.context() as mpatch:
+        # pylint: disable=protected-access
+        # mock the ctypes.CDLL using mock_ctypes_cdll_return()
+        mpatch.setattr(ctypes, "CDLL", mock_ctypes_cdll_return)
+
+        faked_libgmt1 = FakedLibGMT("/path/to/faked/libgmt1.so")
+        faked_libgmt2 = FakedLibGMT("/path/to/faked/libgmt2.so")
+
+        # case 1: two broken libraries
+        # Raise the GMTCLibNotFoundError exception
+        # The error message should contains information of both libraries
+        lib_fullnames = [faked_libgmt1, faked_libgmt2]
+        msg_regex = (
+            fr"Error loading the GMT shared library '{faked_libgmt1._name}'.\n"
+            fr"Error loading '{faked_libgmt1._name}'. Couldn't access.*\n"
+            fr"Error loading the GMT shared library '{faked_libgmt2._name}'.\n"
+            fr"Error loading '{faked_libgmt2._name}'. Couldn't access.*"
+        )
+        with pytest.raises(GMTCLibNotFoundError, match=msg_regex):
+            load_libgmt(lib_fullnames=lib_fullnames)
+
+        # case 2: broken library + invalid path
+        lib_fullnames = [faked_libgmt1, "/invalid/path/to/libgmt.so"]
+        msg_regex = (
+            fr"Error loading the GMT shared library '{faked_libgmt1._name}'.\n"
+            fr"Error loading '{faked_libgmt1._name}'. Couldn't access.*\n"
+            "Error loading the GMT shared library '/invalid/path/to/libgmt.so'.\n"
+            "Unable to find '/invalid/path/to/libgmt.so'"
+        )
+        with pytest.raises(GMTCLibNotFoundError, match=msg_regex):
+            load_libgmt(lib_fullnames=lib_fullnames)
+
+        # case 3: broken library + invalid path + working library
+        lib_fullnames = [faked_libgmt1, "/invalid/path/to/libgmt.so", loaded_libgmt]
+        assert check_libgmt(load_libgmt(lib_fullnames=lib_fullnames)) is None
+
+        # case 4: invalid path + broken library + working library
+        lib_fullnames = ["/invalid/path/to/libgmt.so", faked_libgmt1, loaded_libgmt]
+        assert check_libgmt(load_libgmt(lib_fullnames=lib_fullnames)) is None
+
+        # case 5: working library + broken library + invalid path
+        lib_fullnames = [loaded_libgmt, faked_libgmt1, "/invalid/path/to/libgmt.so"]
+        assert check_libgmt(load_libgmt(lib_fullnames=lib_fullnames)) is None
+
+        # case 6: repeated broken library + working library
+        lib_fullnames = [faked_libgmt1, faked_libgmt1, loaded_libgmt]
+        assert check_libgmt(load_libgmt(lib_fullnames=lib_fullnames)) is None
 
 
 ###############################################################################
```

Comment:
To be honest, I kind of wanted you to move the tests for clib_full_names to a separate file during #872 (e.g. to 'test_clib_loading_full_names'?), wondering if we should do it here since this test file is getting really complicated.

Reply:
I'm OK with moving the tests into separate files.

Reply:
Perhaps still keep the tests in the same file, because it's only ~300 lines, so still maintainable.

I've used comment lines to group the tests:
```
###############################################################################
# Tests for clib_full_names
```